### PR TITLE
Api

### PR DESCRIPTION
**Pull request by miljan-kolosek**

### CODE REVIEW

File: app/controllers/api/v1/stories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      end
+
+      private
+
+        # Access With URL parameter
+        #def restrict_access
+         # api_key = ApiKey.find_by_access_token(params[:access_token])
+         # head :unauthorized unless api_key
+        #end
+
```
Comment (tamara-kolosek) [inline app/controllers/api/v1/stories_controller.rb:36]: delete sufficient comments

File: app/controllers/stories_controller.rb [hunk 2]
```diff
@@ -5,11 +5,10 @@ class StoriesController < ApplicationController
...
-
-      params[:term] ? @stories = Story.all.order("created_at DESC").search(params[:term]) : @stories = Story.all.order("created_at DESC")
-
-      @stories = params[:term] ? Story.all.order("created_at DESC").search(params[:term]) : Story.all.order("created_at DESC")
-
+       @stories = params[:term] ? Story.all.order("created_at DESC").search(params[:term]) : Story.all.order("created_at DESC")
+        respond_to do |format|
+          format.html
+          format.json { render json: @stories }
     else
```
Comment (tamara-kolosek) [inline app/controllers/stories_controller.rb:12]: Why didnt you move needed logic into API controler

File: app/models/api_key.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class ApiKey < ApplicationRecord
+
+  before_create :generate_access_token
+
+  private
+
+  def generate_access_token
+    begin
+      self.access_token = SecureRandom.hex
+    end while self.class.exists?(access_token: access_token)
```
Comment (tamara-kolosek) [inline app/models/api_key.rb:10]: did you find this somewhere?

**Review by tamara-kolosek — CHANGES_REQUESTED** [review]

File: app/models/api_key.rb (hunk 3 above)
Comment (miljan-kolosek) [reply app/models/api_key.rb:10]: I found it through the instruction on the ticket.

File: app/controllers/stories_controller.rb (hunk 2 above)
Comment (miljan-kolosek) [reply app/controllers/stories_controller.rb:12]: I followed the tutorial.

Comment (tamara-kolosek) [reply app/controllers/stories_controller.rb:12]: Ok but if someone will use the api, will get different data?

Comment (miljan-kolosek) [reply app/controllers/stories_controller.rb:12]: Sorry, I don't understand. Can you explain me more about that?

Comment (tamara-kolosek) [reply app/controllers/stories_controller.rb:12]: Just compare your index method in api, and the one in stories controller

**Review by tamara-kolosek — CHANGES_REQUESTED** [review]

File: app/controllers/stories_controller.rb (hunk 2 above)
Comment (miljan-kolosek) [reply app/controllers/stories_controller.rb:12]: @all_stories = Story.all
respond_to do |format|
 format.html
 format.json { render json: @all_stories } 

Is that correct?

File: app/controllers/stories_controller.rb [hunk 4]
```diff
@@ -4,15 +4,17 @@ class StoriesController < ApplicationController
   before_action :set_story, only: [:edit, :show, :update, :destroy]
 
   def index
+    @all_stories = Story.all
```
Comment (tamara-kolosek) [inline app/controllers/stories_controller.rb:7]: why this here. How many queries will you make?

**Review by tamara-kolosek — CHANGES_REQUESTED** [review]

File: app/controllers/stories_controller.rb (hunk 2 above)
Comment (miljan-kolosek) [reply app/controllers/stories_controller.rb:9]: What exactly way do you mean,  I need to cover both cases?

Comment (tamara-kolosek) [reply app/controllers/stories_controller.rb:9]: I just mean on that, if you are doing this search in this controller, why you ommited it in your api version?

Comment (miljan-kolosek) [reply app/controllers/stories_controller.rb:9]: I understood that all stories should be shown in the api version, my mistake. Next step is to change api controller to be same like this controller?

Comment (tamara-kolosek) [reply app/controllers/stories_controller.rb:9]: Do however you want, this is the exercise anyway, but I do not see the reason why that two will behave different.

**Review by tamara-kolosek — APPROVED** [review]

File: app/controllers/stories_controller.rb (hunk 2 above)
Comment (miljan-kolosek) [reply app/controllers/stories_controller.rb:9]: I will change to be the same.

**Pull request merged**